src/panels/lovelace/editor/card-editor/hui-card-editor.ts

```diff
@@ -0,0 +1,264 @@
+import {
+  html,
+  css,
+  LitElement,
+  TemplateResult,
+  CSSResult,
+  customElement,
+  property,
+} from "lit-element";
+
+import yaml from "js-yaml";
+
+import "@material/mwc-button";
+import { HomeAssistant } from "../../../../types";
+import { LovelaceCardConfig } from "../../../../data/lovelace";
+import { LovelaceCardEditor } from "../../types";
+import { getCardElementTag } from "../../common/get-card-element-tag";
+
+import "../../components/hui-yaml-editor";
+// This is not a duplicate import, one is for types, one is for element.
+// tslint:disable-next-line
+import { HuiYamlEditor } from "../../components/hui-yaml-editor";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { EntityConfig } from "../../entity-rows/types";
+
+declare global {
+  interface HASSDomEvents {
+    "entities-changed": {
+      entities: EntityConfig[];
+    };
+    "config-changed": {
+      config: LovelaceCardConfig;
+    };
+  }
+}
+
+@customElement("hui-card-editor")
+export class HuiCardEditor extends LitElement {
+  @property() public hass?: HomeAssistant;
+  @property() public value?;
+
+  @property() private _yaml?: string | undefined;
+  @property() private _configElement?: LovelaceCardEditor;
+  @property() private _configElType?: string;
+  @property() private _GUImode: boolean = true;
+  // Error: Configuration broken - do not save
+  @property() private _error?: string;
+  // Warning: GUI editor can't handle configuration - ok to save
+  @property() private _warning?: string;
+  @property() private _loading: boolean = false;
+
+  public get yaml(): string {
+    return this._yaml || "";
+  }
+  public set yaml(_yaml: string) {
+    this._yaml = _yaml;
+    this._updateConfigElement();
+    // TODO: Defer this to next redraw? How?
```

Comment:
RFC!

Reply:
To defer something after a draw:
```ts
await this.updateComplete
```

Reply:
But you cannot do that in a setter since that's not async. However, since `this.updateComplete` is a promise, you can always `.then` it.

Reply:
Alternative is that you could check inside `updated` if `_yaml` has been updated, and if so, call this.

Reply:
Ok. This just doesn't want to work unless I use this 1ms delay... Tried `updateComplete.then`, tried `updated()`, it all breaks...